src/App.test.tsx

```diff
@@ -1,208 +1,186 @@
-    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`;
+    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`
 
-    renderApp();
+    renderApp()
 
     // The user has logged in but has not granted consent, so check for the consent text
-    await screen.findByText(/requests permission/);
+    await screen.findByText(/requests permission/)
 
     // No redirect should have happened
-    expect(window.location.replace).not.toHaveBeenCalled();
-  });
+    expect(window.location.replace).not.toHaveBeenCalled()
+  })
 
-  test("Consent to app terms", async () => {
-    const consentSpy = jest.spyOn(SynapseClient, "consentToOAuth2Request");
+  test('Consent to app terms', async () => {
+    const consentSpy = jest.spyOn(SynapseClient, 'consentToOAuth2Request')
 
     // Need a token in the cookie so the app tries to use it
-    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`;
+    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`
 
-    renderApp();
+    renderApp()
 
     // The user has logged in but has not granted consent, so check for the consent text
-    await screen.findByText(/requests permission/);
+    await screen.findByText(/requests permission/)
 
-    const consentButton = await screen.findByRole("button", { name: "Allow" });
-    await userEvent.click(consentButton);
+    const consentButton = await screen.findByRole('button', { name: 'Allow' })
+    await userEvent.click(consentButton)
 
     // Should redirect
     // TODO: Verify the redirect URL
-    await waitFor(() => expect(window.location.replace).toHaveBeenCalled());
-    expect(consentSpy).toHaveBeenCalled();
-  });
-  test("Deny consent to app terms", async () => {
-    const consentSpy = jest.spyOn(SynapseClient, "consentToOAuth2Request");
+    await waitFor(() => expect(window.location.replace).toHaveBeenCalled())
+    expect(consentSpy).toHaveBeenCalled()
+  })
+  test('Deny consent to app terms', async () => {
+    const consentSpy = jest.spyOn(SynapseClient, 'consentToOAuth2Request')
     // Need a token in the cookie so the app tries to use it
-    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`;
+    document.cookie = `${ACCESS_TOKEN_COOKIE_KEY}=someToken`
 
-    renderApp();
+    renderApp()
 
     // The user has logged in but has not granted consent, so check for the consent text
-    await screen.findByText(/requests permission/);
+    await screen.findByText(/requests permission/)
 
-    const denyButton = await screen.findByRole("button", { name: "Deny" });
-    await userEvent.click(denyButton);
+    const denyButton = await screen.findByRole('button', { name: 'Deny' })
+    await userEvent.click(denyButton)
 
     // Should redirect
     // TODO: Verify the redirect URL
-    await waitFor(() => expect(window.location.replace).toHaveBeenCalled());
-    expect(consentSpy).not.toHaveBeenCalled();
-  });
-  test("Redirects if a token is provided and the user has already consented, and prompt is not consent", async () => {
```

Comment:
I know the main change in this PR changes this flow, so this might be a lot harder to test. Is there a way we can keep this test? Or at least add it to the rest of the `test.todo`s at the bottom?

Reply:
Added todo